components/suggest/src/rs.rs

```diff
@@ -132,109 +126,81 @@ pub(crate) struct DownloadedAmpSuggestion {
-            .ok_or_else(|| FromSqlError::OutOfRange(v))
-    }
-}
-
-impl SuggestionProvider {
-    #[inline]
-    pub fn from_u8(v: u8) -> Option<Self> {
-        match v {
-            1 => Some(SuggestionProvider::Amp),
-            2 => Some(SuggestionProvider::Wikipedia),
-            _ => None,
-        }
-    }
-}
-
-impl ToSql for SuggestionProvider {
-    fn to_sql(&self) -> RusqliteResult<ToSqlOutput<'_>> {
-        Ok(ToSqlOutput::from(*self as u8))
-    }
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq)]
+pub(crate) struct DownloadedWikipediaSuggestion {
+    #[serde(flatten)]
+    pub common_details: DownloadedSuggestionCommonDetails,
+    #[serde(rename = "icon")]
+    pub icon_id: String,
 }
 
-/// A suggestion to ingest from a downloaded Remote Settings attachment.
+/// A suggestion to ingest from an AMP-Wikipedia attachment downloaded from
+/// Remote Settings.
 #[derive(Clone, Debug, Eq, PartialEq)]
-pub(crate) enum DownloadedSuggestion {
+pub(crate) enum DownloadedAmpWikipediaSuggestion {
     Amp(DownloadedAmpSuggestion),
-    Wikipedia(DownloadedSuggestionCommonDetails),
+    Wikipedia(DownloadedWikipediaSuggestion),
 }
 
-impl DownloadedSuggestion {
-    /// Returns the suggestion fields that are common to AMP and
-    /// Wikipedia suggestions.
+impl DownloadedAmpWikipediaSuggestion {
+    /// Returns the details that are common to AMP and Wikipedia suggestions.
     pub fn common_details(&self) -> &DownloadedSuggestionCommonDetails {
         match self {
             Self::Amp(DownloadedAmpSuggestion { common_details, .. }) => common_details,
-            Self::Wikipedia(common_details) => common_details,
+            Self::Wikipedia(DownloadedWikipediaSuggestion { common_details, .. }) => common_details,
         }
     }
 
+    /// Returns the provider of this suggestion.
     pub fn provider(&self) -> SuggestionProvider {
         match self {
-            DownloadedSuggestion::Amp(_) => SuggestionProvider::Amp,
-            DownloadedSuggestion::Wikipedia(_) => SuggestionProvider::Wikipedia,
+            DownloadedAmpWikipediaSuggestion::Amp(_) => SuggestionProvider::Amp,
+            DownloadedAmpWikipediaSuggestion::Wikipedia(_) => SuggestionProvider::Wikipedia,
         }
     }
 }
 
-impl<'de> Deserialize<'de> for DownloadedSuggestion {
-    fn deserialize<D>(deserializer: D) -> std::result::Result<DownloadedSuggestion, D::Error>
+impl<'de> Deserialize<'de> for DownloadedAmpWikipediaSuggestion {
+    fn deserialize<D>(
+        deserializer: D,
+    ) -> std::result::Result<DownloadedAmpWikipediaSuggestion, D::Error>
     where
         D: Deserializer<'de>,
     {
-        // AMP and Wikipedia suggestions conform to the same JSON schema, but
-        // we want to represent them separately. To distinguish between the two,
-        // we use an "untagged" outer enum and a "tagged" inner enum.
+        // AMP and Wikipedia suggestions are stored in the same attachment
```

Comment:
To clarify, we don't mix those two suggestions within the same RS attachment, and that's enforced at the publishing time.

Reply:
Ooh, thanks for clarifying that; I actually wasn't sure if that was enforced, so I'll update the comment! It doesn't change the logic, but let's make sure the comment is accurate.